mypyc/irbuild/specialize.py

```diff
@@ -219,6 +231,58 @@ def gen_inner_stmts() -> None:
     return retval
 
 
+@specialize_function('builtins.sum')
+def translate_sum_call(builder: IRBuilder, expr: CallExpr, callee: RefExpr) -> Optional[Value]:
+    # specialized implementation is used if:
+    # - only one or two arguments given (if not, sum() has been given invalid arguments)
+    # - first argument is a Generator (there is no benefit to optimizing the performance of eg.
+    #   sum([1, 2, 3]), so non-Generator Iterables are not handled)
+    if not (len(expr.args) in (1, 2)
+        and expr.arg_kinds[0] == ARG_POS
+        and isinstance(expr.args[0], GeneratorExpr)
+    ):
+        return None
+    # handle 'start' argument, if given
+    if len(expr.args) == 2:
+        # ensure call to sum() was improperly constructed
+        if not expr.arg_kinds[1] in (ARG_POS, ARG_NAMED):
+            return None
+        start_expr = expr.args[1]
+        if isinstance(start_expr, IntExpr):
+            target_type = int_rprimitive
+        elif isinstance(start_expr, FloatExpr):
+            target_type = float_rprimitive
+        else:
+            target_type = object_rprimitive
+        # give up if start_expr is not a literal
```

Comment:
I did something yesterday which involved a while loop until there was a .value attribute present, but I can't figure out exactly what right now, and it would never finish for CallCs I think